libraries/src/AWS.Lambda.Powertools.Idempotency/IdempotentAttribute.cs

```diff
@@ -0,0 +1,124 @@
+///         </listheader>
+///         <item>
+///             <term>AWS_LAMBDA_FUNCTION_NAME</term>
+///             <description>string, function name</description>
+///         </item>
+///         <item>
+///             <term>AWS_REGION</term>
+///             <description>string, AWS region</description>
+///         </item>
+///         <item>
+///             <term>POWERTOOLS_IDEMPOTENCY_DISABLED</term>
+///             <description>string, Enable or disable the Idempotency</description>
+///         </item>
+///     </list>
+/// </summary>
+[AttributeUsage(AttributeTargets.Method)]
+[Injection(typeof(UniversalWrapperAspect), Inherited = true)]
+public class IdempotentAttribute : UniversalWrapperAttribute
+{
+    /// <summary>
+    ///     Wraps as a synchronous operation, simply throws IdempotencyConfigurationException
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="target">The target.</param>
+    /// <param name="args">The arguments.</param>
+    /// <param name="eventArgs">The <see cref="AspectEventArgs" /> instance containing the event data.</param>
+    /// <returns>T.</returns>
+    protected internal sealed override T WrapSync<T>(Func<object[], T> target, object[] args, AspectEventArgs eventArgs)
+    {
+        if (PowertoolsConfigurations.Instance.IdempotencyDisabled)
+        {
+            return base.WrapSync(target, args, eventArgs);
+        }
+        var payload = JsonDocument.Parse(JsonSerializer.Serialize(args[0]));
+        if (payload == null)
+        {
+            throw new IdempotencyConfigurationException("Unable to get payload from the method. Ensure there is at least one parameter or that you use @IdempotencyKey");
+        }
+
+        Task<T> ResultDelegate() => Task.FromResult(target(args));
+
+        var idempotencyHandler = new IdempotencyAspectHandler<T>(ResultDelegate, eventArgs.Method.Name, payload);
+        if (idempotencyHandler == null)
+        {
+            throw new Exception("Failed to create an instance of IdempotencyAspectHandler");
+        }
+        var result = idempotencyHandler.Handle().GetAwaiter().GetResult();
+        return result;
+    }
+
+    /// <summary>
+    ///     Wrap as an asynchronous operation.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="target">The target.</param>
+    /// <param name="args">The arguments.</param>
+    /// <param name="eventArgs">The <see cref="AspectEventArgs" /> instance containing the event data.</param>
+    /// <returns>A Task&lt;T&gt; representing the asynchronous operation.</returns>
+    protected internal sealed override async Task<T> WrapAsync<T>(
+        Func<object[], Task<T>> target, object[] args, AspectEventArgs eventArgs)
+    {
+        if (PowertoolsConfigurations.Instance.IdempotencyDisabled)
+        {
+            return await base.WrapAsync(target, args, eventArgs);
+        }
+        var payload = JsonDocument.Parse(JsonSerializer.Serialize(args[0]));
```

Comment:
Check args has value otherwise it raises IndexOutOfRangeException, I for example something like:

```C#
 var payload = args is not null && args.Any()
            ? JsonDocument.Parse(JsonSerializer.Serialize(args.First()))
            : null;
```